db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java

```diff
@@ -328,10 +328,7 @@ else if (operand instanceof ExtractOperatorNode) {
                     if (functionName.equals("SECOND") || functionName.equals("WEEK")) {
                         throwNotImplementedError();
                     } else if (functionName.equals("WEEKDAY") || functionName.equals("DAYOFWEEK_ISO")) {
-                        if( getSparkVersion().getMajorVersionNumber() >= 3 )
-                            return format("(weekday(%s)+1)", opToString2(uop.getOperand(), vars));
-                        else
-                            return format("cast(date_format(%s, \"u\") as int) ", opToString2(uop.getOperand(), vars));
+                        return format("(weekday(%s)+1)", opToString2(uop.getOperand(), vars));
```

Comment:
We still have the hdp3.1.5 platform tied to Spark 2.3, it doesn't look like it would work there, would it?

Reply:
Please consider the implementation in branch DB-11145-jp .  I ran ExternalTableIT on it successfully in my standalone for dbaas3.0, cdh6.3.0, and hdp3.1.5.  See OperatorToString, db-engine/pom.xml, and db-engine/src/main/spark*

Reply:
Opened PR https://github.com/splicemachine/spliceengine/pull/5215 for the implementation in branch DB-11145-jp.